packages/core/src/mm/actors/Player.c

```diff
@@ -1644,3 +1738,437 @@ void Player_UseItem(GameState_Play* play, Actor_Player* this, s16 itemId)
         Player_UseItemImpl(play, this, itemId);
     }
 }
+
+/* Hammer Stuff */
+
+s32 Player_CustomActionToModelGroup(Actor_Player* player, s32 itemAction) {
+    u8* sActionModelGroups = (u8*)0x801BFF3C; /* using original table also means original glitches, if that matters */
+    s32 modelGroup = sActionModelGroups[itemAction];
+
+    if (itemAction == PLAYER_CUSTOM_IA_HAMMER) {
+        modelGroup = sActionModelGroups[7]; /* uses deku stick model group but does not draw deku stick because of the way the original draw code for it works */
+    }
```

Comment:
Maybe this would read better as 
```suggestion
    if (itemAction == PLAYER_CUSTOM_IA_HAMMER) {
        itemAction = PLAYER_IA_DEKU_STICK;
    }
    s32 modelGroup = sActionModelGroups[itemAction];
```
?

Reply:
since the index is fixed I might as well skip using the table and return the model group directly
```
    if (itemAction == PLAYER_CUSTOM_IA_HAMMER) return 10;

    u8* sActionModelGroups = (u8*)0x801BFF3C;
    s32 modelGroup = sActionModelGroups[itemAction];
      if ((modelGroup == 2) && (player->transformation == MM_PLAYER_FORM_GORON || player->transformation == MM_PLAYER_FORM_DEKU)) {
            return 1;
      }
    return modelGroup;
```